src/persist-client/src/batch.rs

```diff
@@ -1189,6 +1201,23 @@ impl Deref for PartDeletes {
     }
 }
 
+/// Returns the total sum of diffs or None if there were no updates.
+fn diffs_sum<'a, D: Semigroup + Codec64>(
```

Comment:
Could move the non-empty assertion to this function if you like; looks like all callers do it anyways!

Kind of a shame we can't just sum up the diffs column, since I'd expect that to vectorize nicely. Oh well... should still be plenty fast!

Reply:
I think I'd like to leave the non-empty asserts where they are. To put it here, we'd have to document it as a precondition of calling this method, which essentially just moves the requirement from being fairly obvious at the callsite to being somewhat less visible.

> Kind of a shame we can't just sum up the diffs column, since I'd expect that to vectorize nicely. Should be plenty fast regardless, though.

Ha, I thought about this too! Doable by adding a bulk sum method to `Codec64` (probably with some default impl), but yeah I'd hold off on that until it becomes an issue in practice